Yank/experiment.py

```diff
@@ -400,7 +400,12 @@ def run(self, n_iterations=None):
 
                 # Run simulation for iterations_left or until we have to switch phase.
                 iterations_to_run = min(iterations_left[phase_id], switch_phase_interval)
-                alchemical_phase.run(n_iterations=iterations_to_run)
+                try:
+                    alchemical_phase.run(n_iterations=iterations_to_run)
+                except utils.SimulationNaNError:
+                    # Simulation has NaN'd, this experiment is done, flag phases as done and send error up stack
+                    self._are_phases_completed = [True] * len(self._are_phases_completed)
```

Comment:
I'm not sure this will work. The `Experiment` instance is rebuilt every time you call `ExperimentBuilder._run_experiment()`, so at the next iteration this variable will still try to run.

This is not a problem if `switch_experiment_interval` is not set anyway, and it's a performance issue rather than a functional one.

Reply:
Ah, you're right, I'll need to think of another way to block the experiment. I don't want to spend time on experiments which have failed, so I'll think about this some more

Reply:
I decided to leave this check in place in case something besides the experiment builder runs the experiment (e.g. through the API) because we do expose Experiment as a public class. The `run_experiments` though now pops finished experiments out so they should not be re-run or even re-built.